src/App.vue

```diff
@@ -24,9 +24,18 @@
     <!-- The nav sticky anchor has to always be between the Header and the Content -->
     <div :id="baseNavStickyAnchorId" />
     <slot :isTargetIDE="isTargetIDE">
-      <router-view class="router-content" />
+      <router-view
+        @availableLocales="availableLocales = $event"
```

Comment:
Not a fan of router-views emitting events, as that is a very broad and generic component. 

Implementing a store so the Footer can have access to the available locales, just for this is probably too much work.

Reply:
First I started working implementing a store, I would set the availableLocales in the AppStore since preferredLocale is also there but this data is not the best place for the AppStore because it's a property depending on each page, not the user's preferences across the app...

That's why I change it to use emitting event + props

Reply:
Hm. I don't really see a clear reason to handle this in a different way than we handle the references data, which is also data that changes from page to page. It's possible I'm missing something subtle that makes it different though.

Reply:
I refactored it to use the AppStore instead https://github.com/apple/swift-docc-render/pull/687/commits/bc8589bb5772a9bd6b381bb72c1830e886a14a44 let see what you think of this other option!